com.unity.mobile.notifications/Runtime/iOS/Plugins/UnityNotificationManager.m

```diff
@@ -168,9 +171,12 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNot
                 notificationData = UNNotificationRequestToiOSNotificationData(notification.request);
                 haveNotificationData = YES;
             }
+            else if (notificationData.userInfo == NULL)
+                notificationData.userInfo = (__bridge_retained void*)notificationUserInfo;
```

Comment:
I didn't remember the details. Not quite sure if there is a case that notificationData.userInfo is NULL but notificationUserInfo is not now. IIRC, they're both converted from `notification.request`?

Reply:
It may get nulled above in line 159. Calling a callback now frees the dictionary, so here we restore the dictionary for the second callback.